app/scripts/directives/blocks.js

```diff
@@ -234,3 +246,104 @@ angular
+        $scope.block.content.staticChoices = [
+          'Yes, via email',
+          'Yes, via text',
+          'Yes, via email & text',
+          'No',
+        ];
+      },
+    };
+  });
+
+angular
+  .module('confRegistrationWebApp')
+  .directive('graduationDateQuestion', function () {
+    return {
+      templateUrl: graduationDateQuestionTemplate,
+      restrict: 'E',
+    };
+  });
+
+angular
+  .module('confRegistrationWebApp')
+  .directive('ethnicityQuestion', function () {
+    return {
+      templateUrl: ethnicityQuestionTemplate,
+      restrict: 'E',
+      controller: function ($scope, $timeout) {
+        $scope.otherSentinel = '__other__';
+
+        $scope.block.content.staticChoices = [
+          'American Indian/Native Alaskan',
+          'Arab/Middle Eastern',
+          'Asian/Asian American',
+          'Black/African American',
+          'Hispanic/Latino',
+          'Multi-racial/Multi-ethnic',
+          'Pacific Islander/Native Hawaiian',
+          'White/European American',
+        ];
+
+        $scope.selectOtherAnswer = () => {
+          if ($scope.otherAnswer) {
+            $scope.selectedAnswer = $scope.otherSentinel;
+          }
+        };
+
+        $scope.clearOther = () => {
+          if ($scope.selectedAnswer != $scope.otherSentinel)
+            $scope.otherAnswer = '';
+        };
+
+        const answerValue = $scope.answer ? $scope.answer.value : '';
+        if (
+          answerValue &&
+          !_.includes($scope.block.content.staticChoices, answerValue)
+        ) {
+          // An answer is present that isn't in the list of available choices, so it is an "Other" answer
+          $scope.selectedAnswer = $scope.otherSentinel;
+          $scope.otherAnswer = answerValue;
+        } else {
+          $scope.selectedAnswer = answerValue;
+          $scope.otherAnswer = '';
+        }
+
+        // We would use ng-model-options="{ debounce: 1000 }", but that causes the selection to disappear when
+        // rapidly changing the selected option (https://stackoverflow.com/questions/31254291). To work-around,
+        // we perform the debounce ourselves. We're writing our own debounce function instead of using _.debounce
+        // because lodash's implementation uses setTimeout, but we need to use $timeout so that the timeout can
+        // be manually flushed in tests.
+        let currentTimeout = null;
+        const updateAnswerValueDebounced = () => {
+          if (currentTimeout) {
+            // Cancel the previous timeout and start a new one
+            $timeout.cancel(currentTimeout);
+          }
+
+          // Update the answer after one second
+          currentTimeout = $timeout(() => {
+            if ($scope.answer) {
+              $scope.answer.value =
+                $scope.selectedAnswer === $scope.otherSentinel
+                  ? $scope.otherAnswer
+                  : $scope.selectedAnswer;
+            }
+          }, 300);
+        };
```

Comment:
I believe we have the timeout functionailty in multiple places. it might be worth creating a Lib file which handles the timeout in which you pass a callback which runs once the timout has completed.

Reply:
I'm not sure how to do that. Could you show an example?